src/csharp/Grpc.Core.Api/ServerCallContext.cs

```diff
@@ -113,6 +114,12 @@ public WriteOptions WriteOptions
         /// </summary>
         public AuthContext AuthContext => AuthContextCore;
 
+        /// <summary>
+        /// Gets a dictionary that can be used by the various interceptors and handlers of this
+        /// call to store arbitrary state.
+        /// </summary>
+        public IDictionary<object, object> UserSate => UserStateCore;
```

Comment:
The only reason we had the Property and PropertyCore pattern was that we didn't want there to be a breaking change when we created an abstract base class. However, given that this is a new property I don't think this is necessary, just make it a virtual with a default implementation.

Reply:
That does make thing easier.